package-scripts/datadog-agent/postinst

```diff
@@ -57,20 +62,24 @@ if which chkconfig >/dev/null 2>&1; then
   chkconfig --add datadog-agent
 fi
 
-if which invoke-rc.d >/dev/null 2>&1; then
-    invoke-rc.d datadog-agent restart
+if which systemd >/dev/null 2>&1; then
+  systemctl restart datadog-agent.service
 else
-    /etc/init.d/datadog-agent restart
-fi
-RETVAL=$?
-if [ $RETVAL -ne 0 ]; then
-    if [ $RETVAL -eq 3 ]; then
-        # No datadog.conf file is present. The user is probably following
-        # the step-by-step instructions and will add the config file next.
-        exit 0
-    else
-        exit $RETVAL
-    fi
+  if which invoke-rc.d >/dev/null 2>&1; then
```

Comment:
Yes it is, but the problem is the return value. Systemctl seems to swallow the exit code of the service it is running. Though I found a hack (pull coming soon, I'm just finishing reviewing your review)